### DISCUSSION TITLE
Time schedule for v7

### DISCUSSION THREAD

**Discussion by zhang-wenchao** [remix-run/react-router#12011] [General]
This is a very important thing that I need to decide and select for my product planning.

**Comment by sergiodxa** [comment]
From the Remix team:

> We can't give a release date because we don't yet know with 100% certainty what the final "RR v7" API is in some cases.  And even if we did, we still can't give a release date because estimating software is impossible 🙃
> As we narrow in on the final API, we then have to figure out the proper bridge from Remix v2 since it needs to be non-breaking.  And in some cases that means build the thing in v7, then backport it to v2 via a future flag that you can adopt for a non-breaking upgrade.

The recommendation is to use Remix v2 if you plan to use RRv7 as RRv7 will be similar to Remix v2 with the future flags enabled.
